sphinx/util/i18n.py

```diff
@@ -118,3 +118,19 @@ def find_catalog_source_files(locale_dirs, locale, domains=None, gettext_compact
                     catalogs.add(cat)
 
     return catalogs
+
+
+def get_image_filename_for_locale(filename, env, force=False):
```

Comment:
This whole method feels weirdly overloaded to me.

It pretends to work off of a locale, but actually operates on the build environment.  It sometimes does name manipulation, sometimes also checks if a file exists.